tests/test_scenario/test_moving_push_notifications.py

```diff
@@ -0,0 +1,505 @@
+                        state = getattr(
+                            conn, "maintenance_state", MaintenanceState.NONE
+                        )
+                        if state != MaintenanceState.NONE:
+                            maintenance_states_found.add(state)
+
+                print(f"Maintenance states found: {maintenance_states_found}")
+
+                # Check operation status first
+                status_result = self.fault_injector.get_action_status(action_id)
+                operation_status = status_result.get("status", "unknown")
+
+                completed_statuses = [
+                    "failed",
+                    "finished",
+                ]
+                if operation_status in completed_statuses:
+                    print(
+                        f"Operation {action_id} completed with status: "
+                        f"{operation_status}"
+                    )
+                    break
+
+                time.sleep(check_interval)
+
+            except Exception as e:
+                print(f"Error checking operation status: {e}")
+                time.sleep(check_interval)
+        else:
+            print(f"Timeout waiting for {operation_name} operation {action_id}")
+
+        print(
+            f"Maintenance states detected during {operation_name}: "
+            f"{maintenance_states_found}"
+        )
+        return maintenance_states_found
+
+    def _get_cluster_nodes_info(self) -> Dict[str, Any]:
+        """Get cluster nodes information from Redis Enterprise."""
+        try:
+            # Use rladmin status to get node information
+            action = ActionRequest(
+                action_type=ActionType.EXECUTE_RLADMIN_COMMAND,
+                parameters={"rladmin_command": "status", "bdb_id": self.endpoint_config.get("bdb_id")},
+            )
+            result = self.fault_injector.trigger_action(action)
+            start_time = time.time()
+
+            while time.time() - start_time < 30:
+                time.sleep(5)
+                something = self.fault_injector.get_action_status(result.get("action_id"))
+                status = something.get("status")
+
+                print(f"Cluster nodes info: {status}")
+                completed_statuses = [
+                    "failed",
+                    "finished",
+                ]
+                if status in completed_statuses:
+                    print(f"Cluster nodes info: {something}")
+                    break
+            return something
+        except Exception as e:
+            pytest.fail(f"Failed to get cluster nodes info: {e}")
+
+    def _find_target_node_and_empty_node(self) -> Tuple[str, str]:
+        """Find the node with master shards and the node with no shards.
+        
+        Returns:
+            tuple: (target_node, empty_node) where target_node has master shards 
+                   and empty_node has no shards
+        """
+        cluster_info = self._get_cluster_nodes_info()
```

Comment:
Is there's a specific reason why you decided to use rladmin and not the REST API here? There's v1/shards endpoint that returns you information about shards so you can easier find the node with master shards and nodes without shards.

rladmin is more an internal tool, whereas REST API is public so I would suggest to use it for this purposes. Since, rladmin is a plain text we might encounter parsing issues if it will be changed whereas REST API returns standardised JSON object

https://redis.io/docs/latest/operate/rs/references/rest-api/requests/shards/#example-json-body